mod/gc_api/lib/profile.php

```diff
@@ -297,6 +288,810 @@ function get_api_profile($id){
 	return $user;
 }
 
+function profileUpdate($id, $data){
+	global $CONFIG;
+	$response['error'] = 0;
+	$user_entity = getUserFromID($id);
+	if (!$user_entity){
+		$response['error'] = 1;
+		$response['message'] = 'Invalid user id, username, or email';
+		return $response;
+		//return "Not a valid user";
+	}
+	
+	if ($data == ''){
+		$response['error'] = 2;
+		$response['message'] = 'data must be a string representing a JSON object.';
+		return $response;
+	}
+	$userDataObj = json_decode($data, true);
+	if (json_last_error() !== 0){
+		$response['error'] = 2;
+		$response['message'] = 'invalid JSON - data was unable to be parsed';
+		return $response;
+		//return "invalid JSON format of data";
+	}
+	
+	//error_log(json_encode($userDataObj));
+	/*
+{ 
+	"name": {
+		"firstName": "Troy",
+		"lastName": "Lawson"
+	},
+	"title": {
+		"en": "GCconnex King",
+		"fr": "le King"
+	},
+	"classification": {
+		"group": "CS",
+		"level": "03"
+	},
+	"department":{
+		"en": "Treasury Board of Canada Secretariat",
+		"fr":	"Secrétariat Conseil du Trésor du Canada"
+	},
+	"branch":{
+		"en": "Information Management and Technology Directorate",
+		"fr": "Direction générale de la gestion d'information et de la technologie"
+	},
+	"sector":{
+		"en": "Corporate Services Sector",
+		"fr": "Secteur des services ministériels"
+	},
+	"location":{
+		"en": {
+			"street": "140 O'Connor St",
+			"city": "Ottawa",
+			"province": "Ontario",
+			"postalCode": "K1A 0R5",
+			"country": "Canada",
+			"building": "L'Esplanade Laurier",
+			"floor": "6",
+			"officeNum": "06062"
+		},
+		"fr": {
+			"street": "140, rue O'Connor",
+			"city": "Ottawa",
+			"province": "Ontario",
+			"postalCode": "K1A 0R5",
+			"country": "Canada",
+			"building": "L'Esplanade Laurier",
+			"floor": "6",
+			"officeNum": "06062"			
+		}
+	},
+	"phone": "613-979-0315",
+	"mobile": "613-979-0315",
+	"email": "Troy.Lawson@tbs-sct.gc.ca",
+	"secondLanguage": {
+		"firstLang": "en",
+		"secondLang": {
+			"lang": "fr",
+			"writtenComp": {
+				"level": "B",
+				"expire": "2016-12-29"
+			},
+			"writtenExpression": {
+				"level": "C",
+				"expire": "2016-12-29"
+			},
+			"oral": {
+				"level": "B",
+				"expire": "2016-12-29"
+			}
+			
+		}
+	}
+}
+	*
+	*
+	*/
+	foreach ($userDataObj as $field => $value){
+		//error_log('in loop');
+		switch($field){
+			case 'name':
+			elgg_set_ignore_access(true);
+			
+				//error_log(json_encode($value));
+				$nameData = json_decode(json_encode($value), true);
+
+				if (!isset($nameData["firstName"])||!isset($nameData["lastName"])){
+						$response['error'] = 4;
+						$response['message'] = 'invalid data format - missing first or last name';
+						return $response;
+
+				}
+				if (!isset($nameData["firstName"])&&!isset($nameData["lastName"])){
+						$response['error'] = 4;
+						$response['message'] = 'invalid data format - missing first or last name';
```

Comment:
I'm guessing this is supposed to say "...missing first __and__ last names"? Otherwise this part doesn't seem to add anything.